collect_app/src/main/java/org/odk/collect/android/javarosawrapper/JavaRosaFormController.java

```diff
@@ -106,6 +107,12 @@ public JavaRosaFormController(File mediaFolder, FormEntryController fec, File in
         this.mediaFolder = mediaFolder;
         formEntryController = fec;
         this.instanceFile = instanceFile;
+        isEditing = instanceFile != null;
```

Comment:
Here you check only if it's not null but in `hasSaved()` method if it exists too. Was that on purpose?

Reply:
Yeah it's on purpose. The two cases are subtly different: When we create a `FormController`, we only pass an instance file to it if we're loading an existing form. For new forms, we then assign an instance file later (but don't create it until saving) which is why we need to check in `FormSaveViewModel` if the instance file exists or not.